### PR TITLE
libsubprocess: make flux_buffer class private

### PR DESCRIPTION
**Pull request by garlick**
Problem: the flux_buffer class and its watchers contribute 34 functions to the public Flux API but for the most part they are only used by libsubprocess, and aren't documented.  Furthermore, this class may need some work to support flow control in the subprocess server.

This moves `flux_buffer_*` et al into libsubprocess and shortens names to `fbuf_*`, making them private to flux-core.  A few minor problems are addressed, and some unused functionality culled.

`flux-exec(1)` and `flux-job(1)` continue to use fbuf to line-buffer standard input.  I didn't think this was necessarily an example of why this code might be needed in the Flux API since that particular use could be trivially implemented another way.

### CODE REVIEW

File: src/common/libsubprocess/fbuf.c [hunk 1]
```diff
@@ -161,25 +161,6 @@ static void nonfull_transition_check (struct fbuf *fb, bool was_full)
     }
 }
 
-int fbuf_drop (struct fbuf *fb, int len)
```
Comment (chu11) [inline src/common/libsubprocess/fbuf.c]: commit message: have to mention adjust a few callers to use fbuf_read over fbuf_peek + fbuf_drop?  Maybe not a huge deal.

**Review by chu11 — APPROVED** [review]
overall LGTM, just a few nits I found

**Comment by garlick** [comment]
@chu11, ok to set MWP or did you want that internal function name changed?

(And thanks for the review!)

**Comment by chu11** [comment]
Do it! :-)

**Pull request merged**